src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py

```diff
@@ -470,6 +473,15 @@ def _record_compile_classpath(self, classpath, target, outdir):
     with open(path, 'w') as f:
       f.write(text)
 
+  def _get_directory_digest_of_dependencies(self, compile_context):
```

Comment:
This will snapshot the target itself, but not it's dependencies. So would consider renaming this method.

But this is also related to the other comment about analysis files: there is conditional logic to either include/ignore analysis files in the `if not is_incremental:` block elsewhere in this file. We implement disabling incremental compiles by not passing an analysis file to zinc.